client/src/app/components/Forms/Login/index.tsx

```diff
@@ -0,0 +1,124 @@
+'use client'
+import React, { useState } from 'react'
+import { FaRegEye, FaRegEyeSlash } from 'react-icons/fa'
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { loginUserValidation } from '@/app/register/validation';
+import { RESULT_STATUS } from '@/constants';
+import { useRouter } from 'next/navigation';
+import Link from 'next/link';
+import { NextResponse } from 'next/server';
+
+const LoginFormComponent = () => {
+    const router = useRouter()
+    const [showPassword, setShowPassword] = useState(false);
+    const [errorMessage, setErrorMessage] = useState("");
+
+    const togglePasswordVisibility = () => {
+        setShowPassword(!showPassword);
+    };
+    const { register, handleSubmit, formState: { errors } } = useForm({ resolver: yupResolver(loginUserValidation) })
+    const onSubmit = async (data: any) => {
```

Comment:
please provide a valid name for the parameters of the submit function instead of common names.
Also, attach the proper types for them instead of the default type